### ISSUE TITLE
Not able to start the application

### ISSUE THREAD

**Issue by ribeaud** [inosca/ebau#9]
Hi.

I cloned the project and followed the instructions. Unfortunately, I was not able to start the project. It looks like I can't access the `document-merge-service`. What should I do?

Thanks and have a great day!

Christian

**Comment by czosel** [comment]
Hi Christian, thanks for your bug report! can you share some more details like logs, error messages etc.?

**Comment by ribeaud** [comment]
Hi @czosel. I thought, this was clear enough, sorry.

I just run `make start-dev-env` after a clone. And this is what I get:

<img width="1425" alt="inosca" src="https://github.com/inosca/ebau/assets/13186156/9c3776aa-5a7f-4fe2-8925-25ecb02ec608">

**Comment by czosel** [comment]
Thanks for posting the screenshot! Can you try starting the App with the kt_gr config? The demo config hasn’t been updated in a while. If you get the same error, it might be related to some auth issue with the GitHub container registry, which I can try to reproduce on Monday.

**Comment by ribeaud** [comment]
Thanks @czosel. Unfortunately that did not help. I am now getting:
```bash
 => => transferring context: 2B                                                                                                                                                                            0.0s
 => ERROR [ember-ebau] importing cache manifest from acr.run/camac-ng/camac-ng/ember-ebau:master                                                                                                           0.4s
 => CANCELED [ember-caluma-portal build  1/12] FROM docker.io/danlynn/ember-cli:4.12.1@sha256:906a6285302384337c4a2e080a7a3fab1a7a14b72d2ef2e945ef5b6beb52f4b1                                             0.8s
 => => resolve docker.io/danlynn/ember-cli:4.12.1@sha256:906a6285302384337c4a2e080a7a3fab1a7a14b72d2ef2e945ef5b6beb52f4b1                                                                                  0.1s
 => [ember-ebau internal] load build context                                                                                                                                                               0.5s
 => => transferring context: 3.64MB                                                                                                                                                                        0.3s
 => CANCELED [ember-caluma-portal prod 1/3] FROM docker.io/nginxinc/nginx-unprivileged:1.25-alpine@sha256:39d8bcb04de4439ee6f3cfa37bdedf75b912a1df23281cf5d3bea213666cf114                                 0.8s
 => => resolve docker.io/nginxinc/nginx-unprivileged:1.25-alpine@sha256:39d8bcb04de4439ee6f3cfa37bdedf75b912a1df23281cf5d3bea213666cf114                                                                   0.1s
 => ERROR [ember-caluma-portal] importing cache manifest from acr.run/camac-ng/camac-ng/ember-caluma-portal:master                                                                                         0.4s
 => ERROR [db] importing cache manifest from acr.run/camac-ng/camac-ng/db:master                                                                                                                           0.3s
 => [db internal] load build context                                                                                                                                                                       0.1s
 => => transferring context: 1.28kB                                                                                                                                                                        0.0s
 => CANCELED [db 1/2] FROM docker.io/library/postgres:12.9-bullseye@sha256:7984919d5b1d8bd22539118c671555b50cd3bb5fbc568645a49f61e5ab94b8f9                                                                0.5s
 => => resolve docker.io/library/postgres:12.9-bullseye@sha256:7984919d5b1d8bd22539118c671555b50cd3bb5fbc568645a49f61e5ab94b8f9                                                                            0.0s
 => CANCELED [ember-caluma-portal internal] load build context                                                                                                                                             0.3s
 => => transferring context: 3.01MB                                                                                                                                                                        0.3s
 => CACHED [ember-ebau build  2/12] COPY package.json pnpm-*.yaml /myapp/                                                                                                                                  0.0s
 => ERROR [ember-ebau build  3/12] COPY ember/package.json /myapp/ember/                                                                                                                                   0.0s
------
 > [ember-ebau] importing cache manifest from acr.run/camac-ng/camac-ng/ember-ebau:master:
------
------
 > [ember-caluma-portal] importing cache manifest from acr.run/camac-ng/camac-ng/ember-caluma-portal:master:
------
------
 > [db] importing cache manifest from acr.run/camac-ng/camac-ng/db:master:
------
------
 > [ember-ebau build  3/12] COPY ember/package.json /myapp/ember/:
------
failed to solve: failed to compute cache key: failed to calculate checksum of ref kg0d9yx0hklxtwhyrv736w8d0::q4fg5f0hfbdhghs36544l7nt9: "/ember/package.json": not found
Run make loadconfig y/n:
```

**Comment by czosel** [comment]
Hi @ribeaud, I was able to reproduce the issue, and just pushed a fix. Can you try again on the current `main` branch? Just to give you some background on the issue: This repository is a slightly modified copy of our internal repository, which is manually synced every now and then. The error was caused by a reference to a legacy container which is not part of the OSS offering.

**Comment by ribeaud** [comment]
Hi @czosel. Sorry but I am still facing the same authorisation problems:

<img width="1665" alt="Screenshot 2024-03-12 at 15 04 09" src="https://github.com/inosca/ebau/assets/13186156/01fa6279-4898-4e28-923b-7990a92a85a4">

I've chosen `kt_gr` as application. If I now select `demo`, this is what I get:

<img width="1670" alt="Screenshot 2024-03-12 at 15 07 33" src="https://github.com/inosca/ebau/assets/13186156/649f38f5-bc7d-494f-8c19-5dec0bb28214">

**Comment by czosel** [comment]
Hmm, the mentioned images are public on the GitHub container registry, so it's strange that you see authorization issues. I assume you get the same error when you run 
```bash
docker pull ghcr.io/adfinis/document-merge-service:dev
```
If so, the issue seems to be with your docker setup. Maybe the fix suggested in https://stackoverflow.com/a/74656301 helps?

**Comment by czosel** [comment]
Hmm, the mentioned images are public on the GitHub container registry, so it's strange that you see authorization issues. I assume you get the same error when you run 
```bash
docker pull ghcr.io/adfinis/document-merge-service:dev
```
If so, the issue seems to be with your docker setup. Maybe the fix suggested in https://stackoverflow.com/a/74656301 helps?

**Comment by ribeaud** [comment]
Hi @czosel. Thanks a lot, that was it.

**Comment by czosel** [comment]
Great, happy to help! If you have further questions, let us know.

**Comment by ribeaud** [comment]
Hi @czosel. Sorry to disturb you again. But, after having pulled the last changes, I am now getting following error:
```
> [ember-ebau build  2/10] COPY package.json pnpm-*.yaml /myapp/COPY ember-caluma-portal/package.json /myapp/ember-caluma-portal/:
------
failed to solve: failed to compute cache key: failed to calculate checksum of ref kg0d9yx0hklxtwhyrv736w8d0::xg9na2gzkg8revswc9altx11w: "/myapp/COPY": not found
Run make loadconfig y/n: n
```
<img width="1276" alt="Screenshot 2024-03-17 at 16 41 33" src="https://github.com/inosca/ebau/assets/13186156/1cb9c257-8b3f-489e-a4e2-4d951efc4a4a">
IIRC, the choice does not play any role here. The problem happens with any other project.

What should I do to be able to start the application and to see how it works?

Many thanks and have an enjoyable Sunday. Best,

Christian

**Comment by czosel** [comment]
Hi @ribeaud, thanks for the bug report. This turned out to be a bug in the repo sync code that cause a linebreak to be missing in the Dockerfile - i've just pushed a bugfix. You might have to do a hard reset of the repository because fixes in the sync code can rewrite git history (sorry about that!).

**Comment by ribeaud** [comment]
> Hi @ribeaud, thanks for the bug report. This turned out to be a bug in the repo sync code that cause a linebreak to be missing in the Dockerfile - i've just pushed a bugfix. You might have to do a hard reset of the repository because fixes in the sync code can rewrite git history (sorry about that!).

Hi @czosel. Thanks for the fix.

**Comment by ribeaud** [comment]
Hi @czosel. The script went through. Unfortunately, not all the containers started. I got following picture:

<img width="1608" alt="docker" src="https://github.com/inosca/ebau/assets/13186156/f42a1b81-cb0e-44eb-b85b-e9422d6264e5">

I selected `demo` (_0_) and `full` (_0_). The **Django** container logs look as following:
```
conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: connection to server at "db" (172.18.0.8), port 5432 failed: FATAL:  database "demo" does not exist
```
Quickly investigating the logs of all the failing containers, they all seem to have a problem finding the `db`. However we see in the picture above that the db connection is up and running. So maybe I have to adapt an environment variable somewhere?
